### ISSUE TITLE
DBeaver Version 7.3.5.202102151746 Error executing tool Dump database on Linux Mint 20.1 x86_64

### ISSUE THREAD

**Issue by urack** [dbeaver/dbeaver#11414]
DBeaver Version 7.3.5.202102151746   

Error executing tool Dump database mariadb Reason java.lang.NullPointerException 

Linux Mint 20.1 x86_64
Mariadb version Server version: 10.3.25-MariaDB-0ubuntu0.20.04.1 Ubuntu 20.04
mariadb-java-client-2.7.2.jar

anyone have similar case here? Not able to find answer after google for several hours.

Error screen capture:

Here are the error logs

!ENTRY org.jkiss.dbeaver.model 4 0 2021-02-19 07:52:48.062 !MESSAGE !SUBENTRY 1 org.jkiss.dbeaver.model 4 0 2021-02-19 07:52:48.062 !MESSAGE !STACK 0 java.lang.NullPointerException at org.jkiss.dbeaver.ext.mysql.tasks.MySQLExportSettings.setClientHome(MySQLExportSettings.java:311) at org.jkiss.dbeaver.tasks.ui.nativetool.AbstractNativeToolWizard.updateErrorMessage(AbstractNativeToolWizard.java:147) at org.jkiss.dbeaver.tasks.ui.nativetool.AbstractNativeToolWizard.createPageControls(AbstractNativeToolWizard.java:134) at org.eclipse.jface.wizard.WizardDialog.createPageControls(WizardDialog.java:743) at org.eclipse.jface.wizard.WizardDialog.createContents(WizardDialog.java:635) at org.jkiss.dbeaver.tasks.ui.wizard.TaskConfigurationWizardDialog.createContents(TaskConfigurationWizardDialog.java:157) at org.eclipse.jface.window.Window.create(Window.java:431) at org.eclipse.jface.dialogs.Dialog.create(Dialog.java:1094) at org.eclipse.jface.window.Window.open(Window.java:788) at org.jkiss.dbeaver.ext.mysql.tools.MySQLToolExport.execute(MySQLToolExport.java:39) at org.jkiss.dbeaver.tasks.ui.ExecuteToolHandler.executeTool(ExecuteToolHandler.java:59) at org.jkiss.dbeaver.tasks.ui.ExecuteToolHandler.run(ExecuteToolHandler.java:51) at org.jkiss.dbeaver.ui.ActionUtils$1.run(ActionUtils.java:363) at org.eclipse.jface.action.Action.runWithEvent(Action.java:474) at org.eclipse.jface.action.ActionContributionItem.handleWidgetSelection(ActionContributionItem.java:580) at org.eclipse.jface.action.ActionContributionItem.lambda$4(ActionContributionItem.java:414) at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:89) at org.eclipse.swt.widgets.Display.sendEvent(Display.java:5785) at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1427) at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:5048) at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:4526) at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1157) at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338) at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1046) at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:155) at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:644) at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338) at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:551) at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:156) at org.jkiss.dbeaver.ui.app.standalone.DBeaverApplication.start(DBeaverApplication.java:243) at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:203) at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134) at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104) at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:401) at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:255) at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) at java.base/java.lang.reflect.Method.invoke(Unknown Source) at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:653) at org.eclipse.equinox.launcher.Main.basicRun(Main.java:590) at org.eclipse.equinox.launcher.Main.run(Main.java:1461) at org.eclipse.equinox.launcher.Main.main(Main.java:1434)

#### System information: 
- Operating system (distribution) and version
- DBeaver version
- Additional extensions

#### Connection specification:
- Database name and version
- Driver name
- Do you use tunnels or proxies (SSH, SOCKS, etc)?

**Comment by flambchap** [comment]
Hello @urack. I'd like to reproduce the bug, so I need to know which local server you use to make a DB dump. Thank you.

**Comment by urack** [comment]
> Hello @urack. I'd like to reproduce the bug, so I need to know which local server you use to make a DB dump. Thank you.

Thanks for your support. 

Actually i am running Linux Mint 20.1 x86_64 which is based on ubuntu on my laptop to do testing. The database server is Mariadb version Server version: 10.3.25-MariaDB-0ubuntu0.20.04.1 Ubuntu 20.04.   

I installed DBeaver Version 7.3.5.202102151746 on same machine.  If you need any other information , feel free let me know.

**Comment by piotrku** [comment]
The same here:

Error executing tool 'Dump database' 
Reason:
java.lang.NullPointerException

for:
DBeaver Version 7.3.5.202102151746
macOS 11.2.1 (Big Sur)
Driver: MySQL Connector/J [mysql-connector-java-8.0.17 (Revision: 16a712ddb3f826a1933ab42b0039f7fb9eebc6ec)]
Server URL: jdbc:mysql://localhost:33060/
Server Server: MySQL [8.0.23-0ubuntu0.20.04.1]
```

java.lang.NullPointerException
	at org.jkiss.dbeaver.ext.mysql.tasks.MySQLExportSettings.setClientHome(MySQLExportSettings.java:311)
	at org.jkiss.dbeaver.tasks.ui.nativetool.AbstractNativeToolWizard.updateErrorMessage(AbstractNativeToolWizard.java:147)
	at org.jkiss.dbeaver.tasks.ui.nativetool.AbstractNativeToolWizard.createPageControls(AbstractNativeToolWizard.java:134)
	at org.eclipse.jface.wizard.WizardDialog.createPageControls(WizardDialog.java:743)
	at org.eclipse.jface.wizard.WizardDialog.createContents(WizardDialog.java:635)
	at org.jkiss.dbeaver.tasks.ui.wizard.TaskConfigurationWizardDialog.createContents(TaskConfigurationWizardDialog.java:157)
	at org.eclipse.jface.window.Window.create(Window.java:431)
	at org.eclipse.jface.dialogs.Dialog.create(Dialog.java:1094)
	at org.eclipse.jface.window.Window.open(Window.java:788)
	at org.jkiss.dbeaver.ext.mysql.tools.MySQLToolExport.execute(MySQLToolExport.java:39)
	at org.jkiss.dbeaver.tasks.ui.ExecuteToolHandler.executeTool(ExecuteToolHandler.java:59)
	at org.jkiss.dbeaver.tasks.ui.ExecuteToolHandler.run(ExecuteToolHandler.java:51)
	at org.jkiss.dbeaver.ui.ActionUtils$1.run(ActionUtils.java:363)
	at org.eclipse.jface.action.Action.runWithEvent(Action.java:474)
	at org.eclipse.jface.action.ActionContributionItem.handleWidgetSelection(ActionContributionItem.java:580)
	at org.eclipse.jface.action.ActionContributionItem.lambda$4(ActionContributionItem.java:414)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:89)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4443)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1512)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1535)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1520)
	at org.eclipse.swt.widgets.Widget.notifyListeners(Widget.java:1324)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4229)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3839)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1157)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1046)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:155)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:644)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:551)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:156)
	at org.jkiss.dbeaver.ui.app.standalone.DBeaverApplication.start(DBeaverApplication.java:243)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:203)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:401)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:255)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:653)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:590)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1461)
```

**Comment by flambchap** [comment]
Fixed.

**Comment by cqyisbug** [comment]
nice

**Comment by uslss** [comment]
verified

**Comment by sho13** [comment]
Hi, i'm experiencing the same issue, is there any news when the fix will be available? Thank you 🙏!

**Comment by flambchap** [comment]
Hi. The fix is included in version 21.0.0 which will be out on Monday.
